lib/ProductOpener/Products.pm

```diff
@@ -1925,7 +1926,7 @@ sub replace_user_id_in_product ($product_id, $user_id, $new_user_id) {
 	}
 
 	if ((defined $most_recent_product_ref) and (not $most_recent_product_ref->{deleted})) {
-		my $products_collection = get_products_collection();
```

Comment:
Could we make the field mandatory, and not have a default?

Then we would need a change in scripts/find_and_replace_user_id_in_products.pl as well

Reply:
Actually no need to change the script as the script uses the parent, plural sub